core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider_life_cycle.go

```diff
@@ -17,30 +17,62 @@ var (
 	LogRetention = 24 * time.Hour
 )
 
-func (p *logEventProvider) RegisterFilter(upkeepID *big.Int, cfg LogTriggerConfig) error {
+func (p *logEventProvider) RegisterFilter(opts FilterOptions) error {
+	upkeepID, cfg := opts.UpkeepID, opts.TriggerConfig
 	if err := p.validateLogTriggerConfig(cfg); err != nil {
 		return errors.Wrap(err, "invalid log trigger config")
 	}
-	filter := p.newLogFilter(upkeepID, cfg)
+	lpFilter := p.newLogFilter(upkeepID, cfg)
 
-	if p.filterStore.Has(upkeepID) {
-		// TODO: check if filter was updated, if so unregister previous filter and register new one
-		return errors.Errorf("filter for upkeep with id %s already registered", upkeepID.String())
+	// using lock to facilitate multiple events causing filter registration
+	// at the same time.
+	// TODO: consider using a q to handle registration requests
```

Comment:
nit. not sure if you are planning to do this before launch, if not can you update to exploratory